oak-store-spi/src/main/java/org/apache/jackrabbit/oak/spi/filter/PathFilter.java

```diff
@@ -195,4 +197,12 @@ public boolean areAllDescendantsIncluded(String path) {
         }
         return false;
     }
+
+    private static void checkPathsAreAbsolute(Iterable<String> paths, String pathType) {
+        for (String path : paths) {
+            if (!PathUtils.isAbsolute(path)) {
+                throw new IllegalStateException("Invalid path in " + pathType + " paths list: " + path + ". Paths must be absolute.");
```

Comment:
would it be better to throw an IllegalArgumentException instead ?

Reply:
I agree it would be better, but the PathFilter constructor is already throwing `IllegalStateException` in a check it does in the constructor, so I prefer to keep it consistent and use the same exception.